### PR TITLE
[WIP] Convert testplan page to patternfly

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

File: tcms/testplans/views.py [hunk 1]
```diff
@@ -84,6 +86,66 @@ def post(self, request):
         return render(request, self.template_name, context_data)
 
 
+@require_GET
+def get_all(request):  # pylint: disable=missing-permission-required
```
Comment (atodorov) [inline tcms/testplans/views.py]: This was previously removed, is it needed ?

File: tcms/testplans/templates/testplans/get.html [hunk 2]
```diff
@@ -0,0 +1,229 @@
+{% extends "base.html" %}
+{% load i18n %}
+{% load static %}
+{% load comments %}
+{% load extra_filters %}
+{% load attachments_tags %}
+
+{% block title %}TP-{{ object.pk }}: {{ object.summary }}{% endblock %}
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: test plans don't have summary

File: tcms/testplans/templates/testplans/get.html [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+{% block body_class %}cards-pf{% endblock %}
+
+{% block contents %}
+    <div class="container-cards-pf">
+    <!-- Important:  if you need to nest additional .row within a .row.row-cards-pf, do *not* use .row-cards-pf on the nested .row  -->
+    <h1 class="col-md-12" style="margin-top: 0">
+        <span id="test_case_pk"
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: fix selector name

File: tcms/testplans/templates/testplans/get.html [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            <div class="card-pf card-pf-accented card-pf-aggregate-status" style="text-align: left">
+                <h2 class="card-pf-title">
+                    <span class="fa pficon-user"></span>{% trans 'Author' %}:
+                    <a href="{% url "tcms-profile" object.author.username %}">{{ object.author.username }}</a>
+                </h2>
+
+
+                <h2 class="card-pf-title">
+                <span id="product_pk"
+                      data-pk="{{ object.category.product.pk }}"
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: fix variable name

File: tcms/testplans/templates/testplans/get.html [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                       title="Search plans of {{ object.product }} ">{{ object.product }}</a>
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-calendar"></span>{{ object.create_date }}
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-columns"></span>{% trans 'Version' %}:
+                    {{ object.product_version.value }}
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: move version immediately below product

File: tcms/testplans/templates/testplans/get.html [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+                <span id="product_pk"
+                      data-pk="{{ object.category.product.pk }}"
+                      class="fa fa-shopping-cart"></span>{% trans 'Product' %}:
+                    <a href="{% url "plans-search" %}?product={{ object.product_id }}"
+                       title="Search plans of {{ object.product }} ">{{ object.product }}</a>
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-calendar"></span>{{ object.create_date }}
+                </h2>
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: Move this after Author

File: tcms/testplans/templates/testplans/get.html [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-columns"></span>{% trans 'Version' %}:
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: use `fa-code-fork` for version icon

File: tcms/testplans/templates/testplans/get.html [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+                    <span class="fa fa-cube"></span>{% trans 'Plan Type' %}:
+                    {{ object.type.name  }}
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-arrow-circle-o-up"></span>{% trans 'Parent' %}:
+                    {% if object.parent.pk %}
+                        <a href="{% url 'test_plan_url_short' object.parent.pk %}">{{ object.parent.pk }}</a>
+                    {% else %}
+                        {% trans "None" %}
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: Substitue with `-` for missing value like we have for reference link

File: tcms/testplans/templates/testplans/get.html [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+{#        </div>#}
+{##}
+{#    </div><!-- /row -->#}
+</div>
+<script src="{% static 'typeahead.js/dist/typeahead.jquery.min.js' %}"></script>
+
+<script src="{% static 'js/utils.js' %}"></script>
+<script src="{% static 'js/jsonrpc.js' %}"></script>
+<script src="{% static 'js/tags.js' %}"></script>
+<script src="{% static 'testcases/js/get.js' %}"></script>
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: Not needed

File: tcms/testplans/templates/testplans/get.html [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+                        <a href="{{ object.extra_link }}">{{ object.extra_link }}</a>
+                    {% else %}
+                        -
+                    {% endif %}
+                </h2>
+                <div class="card-pf-body"></div>
+            </div>
+        </div>
+    </div>
+{##}
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: IMO drop the entire commented section and start building it widget by widget. If you remove it then we can start merging into the working branch.

File: tcms/testplans/templates/testplans/get.html [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+                    <a href="{% url "tcms-profile" object.author.username %}">{{ object.author.username }}</a>
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-calendar"></span>{{ object.create_date }}
+                </h2>
+
+                <h2 class="card-pf-title">
+                <span id="product_pk"
+                      data-product-pk="{{ object.category.product.pk }}"
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: TestPlan doesn't have category, it does have Product. Not sure why this works, probably this is an auto-generated related objects manager.

File: tcms/testplans/templates/testplans/get.html (hunk 7 above)
Comment (atodorov) [reply tcms/testplans/templates/testplans/get.html]: I see the new icon and remember an issue I had when designing the TC page. Some icons are thinner than others and the text doesn't align with what is above it. You

File: tcms/testplans/templates/testplans/get.html [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-cube"></span>{% trans 'Plan Type' %}:
+                    {{ object.type.name  }}
+                </h2>
+
+                <h2 class="card-pf-title">
+                    <span class="fa fa-arrow-circle-o-up"></span>{% trans 'Parent' %}:
+                    {% if object.parent.pk %}
+                        <a href="{% url 'test_plan_url_short' object.parent.pk %}">{{ object.parent.pk }}</a>
```
Comment (atodorov) [inline tcms/testplans/templates/testplans/get.html]: That should be TP-XX

File: tcms/testplans/templates/testplans/get.html (hunk 11 above)
Comment (RMadjev) [reply tcms/testplans/templates/testplans/get.html]: It doesnt work "<span id="product_pk" **data-product-pk=""** class="fa fa-shopping-cart"></span>", I will remove the data attribute and if add it later on if I needed it.

**Pull request merged**